Assets/Scripts/PlayerRotation.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.Rendering;
+
+/// <summary>
+/// Controls the rotation of a player object. Tilts the camera for vertical rotation, spins the character controller for horizontal rotation.
+/// Locks the cursor if configured to do so
+/// </summary>
+[RequireComponent(typeof(CharacterController))]
+public class PlayerRotation : MonoBehaviour
+{
+    [SerializeField][Range(45, 90)] float maxYaw;  
+    [SerializeField][Range(1, 100)] float lookSensitivity;
+    [SerializeField][Range(0.5f, 1)] float verticalModifier;
+    [SerializeField] bool lockCursor;
+    private InputAction lookAction;
+    private CharacterController cc;
+    private Camera cam;
+    private float yaw;
+
+    void Awake()
+    {
+        Cursor.lockState = lockCursor ? CursorLockMode.Locked : CursorLockMode.None;
+        
+        cc = GetComponent<CharacterController>();
+        cam = GetComponentInChildren<Camera>();
+    }
+
+    void Start()
+    {
+        lookAction = InputSystem.actions.FindAction("Look");
+    }
+
+    void Update()
+    {
+        Vector2 lookInput = lookAction.ReadValue<Vector2>() * lookSensitivity;
+
+        // rotate character horizontally (around the Y axis)
+        transform.Rotate(Vector3.up * lookInput.x * Time.deltaTime);
+
+        // rotate camera vertically (around the X axis) with clamp
+        yaw += -lookInput.y * verticalModifier * Time.deltaTime;
+        yaw = Mathf.Clamp(yaw, -maxYaw, maxYaw);
```

Comment:
Good thinking! Accidentally rotating the camera too far would be quite frustrating when you're fighting your mouse and player just to see straight!